src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -93,32 +99,53 @@ - (void)removePeripheralsFromCache;
 
         void BleConnectionDelegateImpl::NewConnection(
             Ble::BleLayer * bleLayer, void * appState, const SetupDiscriminator & deviceDiscriminator)
+        {
+            NewConnectionImpl(bleLayer, appState, chip::Span(&deviceDiscriminator, 1),
+                OnConnectionComplete, nullptr, OnConnectionError);
+        }
+
+        CHIP_ERROR BleConnectionDelegateImpl::NewConnection(BleLayer * bleLayer, void * appState, const Span<const SetupDiscriminator> & discriminators,
+            OnConnectionByDiscriminatorsCompleteFunct onConnectionComplete,
+            OnConnectionErrorFunct onConnectionError)
+        {
+            NewConnectionImpl(bleLayer, appState, discriminators, nullptr, onConnectionComplete, onConnectionError);
+            return CHIP_NO_ERROR;
+        }
+
+        void BleConnectionDelegateImpl::NewConnectionImpl(BleLayer * bleLayer, void * appState, const chip::Span<const SetupDiscriminator> & desiredDiscriminators,
+            BleConnectionDelegate::OnConnectionCompleteFunct onConnectionComplete,
+            BleConnectionDelegate::OnConnectionByDiscriminatorsCompleteFunct onConnectionCompleteWithDiscriminator,
+            BleConnectionDelegate::OnConnectionErrorFunct onConnectionError)
         {
             assertChipStackLockedByCurrentThread();
 
-            if (deviceDiscriminator.IsShortDiscriminator()) {
-                ChipLogProgress(Ble, "ConnectionDelegate NewConnection with short discriminator %d (0x%x)", deviceDiscriminator.GetShortValue(), deviceDiscriminator.GetShortValue());
-            } else {
-                ChipLogProgress(Ble, "ConnectionDelegate NewConnection with long discriminator %d (0x%x)", deviceDiscriminator.GetLongValue(), deviceDiscriminator.GetLongValue());
+            for (auto & deviceDiscriminator : desiredDiscriminators) {
+                if (deviceDiscriminator.IsShortDiscriminator()) {
+                    ChipLogProgress(Ble, "ConnectionDelegate NewConnection with short discriminator %d (0x%x)", deviceDiscriminator.GetShortValue(), deviceDiscriminator.GetShortValue());
```

Comment:
Can we change this so it doesn't look like `NewConnection` was called multiple times for the multiple discriminator case? Ideally we could just format the discriminators as a list, even though that's a little tedious. Maybe we can just format short and long descriptors with 1 vs 3 hex digits and tell them apart that way?

Reply:
I guess I can use %@ with an NSString here?  I can do better formatting.

Reply:
Yes %@ works in Darwin logging

Reply:
Done.